src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php

```diff
@@ -521,6 +521,41 @@ public function testObjectNormalizerRegistered()
         $this->assertEquals(-1000, $tag[0]['priority']);
     }
 
+    public function testSerializerCacheActivated()
+    {
+        $container = $this->createContainerFromFile('serializer_enabled');
+        $this->assertTrue($container->hasDefinition('serializer.mapping.cache_class_metadata_factory'));
+    }
+
+    public function testSerializerCacheDisabled()
+    {
+        $container = $this->createContainerFromFile('serializer_enabled', array('kernel.debug' => true, 'kernel.container_class' => __CLASS__));
+        $this->assertFalse($container->hasDefinition('serializer.mapping.cache_class_metadata_factory'));
+    }
+
+    public function testDeprecatedSerializerCacheOption()
```

Comment:
Should be tagged `@group legacy` so that we don't forget removing it when preparing 4.0